assets/db_schema/all_entities.iuml

```diff
@@ -0,0 +1,343 @@
+' Zero or One	|o--
+' Exactly One	||--
+' Zero or Many	}o--
+' One or Many	}|--
+' One to Many	||--o{
+' Many to One	}o--||
+
+@startuml(id=swap)
+
+entity "swap" as swap {
+    id : <<pk>>,
+    ---
+    local_swap_id : <<unique>>,
+    role,
+    counter_party_peer_id
+}
+
+entity "shared_swap_id" as shared_swap_id {
+    id : <<pk>> <<fk>>,
+    ---
+    shared_swap_id
+}
+
+entity "address_hint" as address_hint {
+    id : <<pk>>
+    ---
+    counter_party_peer_id : <<unique>> <<fk>>,
```

Comment:
What `fk` is this pointing to?

Couldn't this just be a table independent of swaps where we just keep track of address<->peerid mappings?

Reply:
At least for Alice the address_hint is mandatory at the moment. That's why we modelled it like hat. 
We could argue that this is an implementation detail and keep it separate from the swap.

Reply:
Yes, the `fk` should be removed. One can only link to primary keys or UNIQUE fields.

Reply:
Not update :( 
I'll keep it locally instead of regenerating all links and will push it once approved.